pulpcore/download/file.py

```diff
@@ -37,6 +37,9 @@ def __init__(self, url, *args, **kwargs):
         RemoteSerializer().validate_url(url)
         p = urlparse(url)
         self._path = os.path.abspath(os.path.join(p.netloc, p.path))
+        if kwargs.get("headers_ready_callback", None):
+            self.headers_ready_callback = kwargs.get("headers_ready_callback")
```

Comment:
It looks like it's perfectly safe to have `headers_ready_callback==None` (also `get` defaults to `None` anyway).
```suggestion
        self.headers_ready_callback = kwargs.get("headers_ready_callback")
```

Reply:
That code turned out to not be the actual issue (alas)